File: frontend/libs/common/src/lib/types/question.ts
import { Role } from '@epam/ai-dial-overlay';

export enum QuestionStatus {
  UNDECIDED = 'UNDECIDED',
  ACCEPTED = 'ACCEPTED',
  DISCARDED = 'DISCARDED',
}

export interface QuestionMetadata {
  question_file: string;
  status: QuestionStatus;
  reviewed: boolean;
  name: string;
}

export interface Question {
  id: string;
  question_file: string;
  name: string;
  timestamp: string;
  status: QuestionStatus;
  reviewed: boolean;
  history: {
    content: string;
    role: Role;
  }[];
  question: string;
  answer: string;
  original_sheets: Record<string, string>;
  solution_sheets: Record<string, string>;
  focus: {
    sheet_name: string;
    table_name: string;
    column_name: string;
  }[];
}

File: python/assistant/src/dial/xl/assistant/dto/question.py
from aidial_sdk.chat_completion import Role
from pydantic import UUID4, AwareDatetime, BaseModel

from dial.xl.assistant.dto.focus import ColumnDTO
from dial.xl.assistant.dto.question_status import QuestionStatus


class HashColumnDTO(ColumnDTO):
    hash: str = ""


class MessageDTO(BaseModel):
    content: str
    role: Role


class QuestionDTO(BaseModel):
    id: UUID4
    timestamp: AwareDatetime
    status: QuestionStatus

    history: list[MessageDTO]
    question: str
    answer: str

    reviewed: bool

    original_sheets: dict[str, str]
    solution_sheets: dict[str, str]

    focus: list[HashColumnDTO]
